### ISSUE TITLE
Exception is not correctly logged when calling without message

### ISSUE THREAD

**Issue by MichaelLogutov** [NLog/NLog#787]
For NLog 4.0.1

Code
```c#
LogManager.GetCurrentClassLogger ().Error (new InvalidOperationException ("Test"), "test");
LogManager.GetCurrentClassLogger ().Error (new InvalidOperationException ("Test"));
```

With NLog.config
```xml
<?xml version="1.0" encoding="utf-8"?>

<nlog xmlns="http://www.nlog-project.org/schemas/NLog.xsd" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
      autoReload="true" throwExceptions="true">

    <targets async="true">

        <target name="Console"
                xsi:type="Console"
                layout="[${date:format=dd.MM.yyyy HH\:mm\:ss}] ${exception:format=tostring}" />

    </targets>

    <rules>
        <logger name="*" writeTo="Console" />
    </rules>

</nlog>
```

Outputs
```
[02.07.2015 15:45:56] System.InvalidOperationException: Test
[02.07.2015 15:45:56]
```

**Comment by 304NotModified** [comment]
That's a serious bug. We will add those to the unit tests (there are many unit tests for error logging already)

**Comment by 304NotModified** [comment]
I found the issue. Without the message argument it's calling the generic version: 

See #788 for details.

**Comment by 304NotModified** [comment]
PR merged and so this is fixed in 4.1

**Comment by 304NotModified** [comment]
Closing as this had been merged
